corehq/warehouse/models/facts.py

```diff
@@ -120,17 +120,17 @@ class ApplicationStatusFact(BaseFact, CustomSQLETLMixin):
     '''
     slug = APP_STATUS_FACT_SLUG
 
-    app_dim = models.ForeignKey(ApplicationDim, on_delete=models.PROTECT)
+    app_dim = models.ForeignKey(ApplicationDim, on_delete=models.PROTECT, null=True)
 
     domain = models.CharField(max_length=255, db_index=True)
 
     user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT)
 
-    last_form_submission_date = models.DateTimeField()
-    last_sync_log_date = models.DateTimeField()
+    last_form_submission_date = models.DateTimeField(null=True)
+    last_sync_log_date = models.DateTimeField(null=True)
 
-    last_form_app_build_version = models.CharField(max_length=255)
-    last_form_app_commcare_version = models.CharField(max_length=255)
+    last_form_app_build_version = models.CharField(max_length=255, null=True)
+    last_form_app_commcare_version = models.CharField(max_length=255, null=True)
```

Comment:
same question about these not null fields. will the process now fail silently if the left joins don't turn up matches, and is that ok?

Reply:
I forgot to account for users who have synced but never submitted forms. Those need to be nullable so those users will turn up in the report (and vice versa for the synclog fields i changed). Nothing should fail silently, it should just insert the null values we want